indexer.ts

```diff
@@ -0,0 +1,743 @@
+          : donation.matching_pool
+          ? "Donate_Pot_Matching_Pool"
+            : "Donate_Pot_Public",
+      action_result: recipient,
+      tx_hash: receiptId,
+    };
+
+    await context.db.Activity.insert(activity);
+  }
+
+  // map through the successful actions and swittch on the methodName called for each, then call the designated handlerFunction.
+  await Promise.all(
+    matchingActions.flatMap((action) => {
+      action.operations.map(async (operation) => {
+        console.log("see the contents here...:,==", operation["FunctionCall"]);
+        let call = operation["FunctionCall"];
+        if (call) {
+          const args = atob(call.args); // decode function call argument
+          switch (call.methodName) {
+            case "new":
+            // new can be called on  a couple of things, if the recever id matches factory pattern, then it's a new factory, else if it matches the registry account id, then it's a registry contract initialization, else, it's a new pot initialization.
+              matchVersionPattern(action.receiverId)
+                ? await handleNewFactory(
+                    args,
+                    action.receiverId,
+                    action.signerId,
+                    action.receiptId
+                  )
+                : action.receiverId == "registry.potlock.near" // initializing registry
+                ? await handleRegistry(
+                    args,
+                    action.receiverId,
+                    action.signerId,
+                    action.receiptId
+                  )
+                : await handleNewPot(
+                    args,
+                    action.receiverId,
+                    action.signerId,
+                    action.predecessorId,
+                    action.receiptId
+                  );
+              break;
+            // this is the callback after user applies to a certain pot, it can either be this or handle_apply
+            case "assert_can_apply_callback":
+              console.log("application case:", JSON.parse(args));
+              await handlePotApplication(
+                args,
+                action.receiverId,
+                action.signerId,
+                action.receiptId
+              );
+              break;
+            case "handle_apply":
+              console.log("application case 2:", JSON.parse(args));
+              break;
+            
+            // if function call is donate, call the handle new donations function
+            case "donate":
+              console.log("donatons to project incoming:", JSON.parse(args));
+              await handleNewDonations(
+                args,
+                action.receiverId,
+                action.signerId,
+                "donate",
+                action.receiptId
+              );
+              break;
+            
+            // this is a form of donation where user calls donate on a pot
+            case "handle_protocol_fee_callback":
```

Comment:
this should probably listen to the `process_donation` callback which is at the end of the callback chain and returns the new donation